packages/core/src/provider/base.ts

```diff
@@ -40,7 +40,7 @@ export const noOpMiddleware: AccountMiddlewareFn = async (
 ) => struct;
 
 // borrowed from ethers.js
-function defineReadOnly<T, K extends keyof T>(
+export function defineReadOnly<T, K extends keyof T>(
```

Comment:
we might want to move this to a `utils` file / folder

Reply:
noted